sdk/core/azure-core-http-jdk-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkHttpClientProvider.java

```diff
@@ -13,9 +14,28 @@
  * introduced.
  */
 public final class JdkHttpClientProvider implements HttpClientProvider {
+    // Enum Singleton Pattern
+    private enum GlobalJdkAsyncHttpClient {
+        HTTP_CLIENT(new JdkAsyncHttpClientBuilder().build());
+
+        private final HttpClient httpClient;
+
+        GlobalJdkAsyncHttpClient(HttpClient httpClient) {
+            this.httpClient = httpClient;
+        }
+
+        private HttpClient getHttpClient() {
+            return httpClient;
+        }
+    }
 
     @Override
     public HttpClient createInstance() {
-        return new JdkAsyncHttpClientBuilder().build();
+        final String disableDefaultSharingHttpClient =
+            Configuration.getGlobalConfiguration().get("AZURE_DISABLE_HTTP_CLIENT_SHARING");
```

Comment:
I think we should consider resolving this eagerly in static context and don't look up every time. I don't think somebody is going to flip this in the middle of process run.